### PR TITLE
Allow reading and writting files in google drive

### PR DESCRIPTION
**Pull request by capitancambio**
With this pr we add
* File creation in google drive
* File updates
* Getting the contents of a file
* Registering all the new functionallity to the apis 
* fix small issue to get to_excel working due to change in zipfile in 3.7

### CODE REVIEW

File: src/tentaclio/__init__.py [hunk 1]
```diff
@@ -52,7 +56,10 @@
 SCANNER_REGISTRY.register("s3", ClientDirScanner(S3Client))
 
 SCANNER_REGISTRY.register("sftp", ClientDirScanner(SFTPClient))
-SCANNER_REGISTRY.register("ftp", ClientDirScanner(FTPClient))
+SCANNER_REGISTRY.register("sftp", ClientDirScanner(SFTPClient))
```
[the remark this answers is not in the document]
Comment (ali-tny) [reply src/tentaclio/__init__.py:59]: ah, functional tests caught it anyway 👍

File: src/tentaclio/clients/google_drive_client.py [hunk 2]
```diff
@@ -91,66 +166,124 @@ def close(self) -> None:
...
+            file_descriptor = self._get_leaf_descriptor()
+            self._update(file_descriptor, reader)
+        except DescriptorNotFound:
+            # file doesn't exist, then create
+            self._create(reader)
+
+    def _create(self, reader: protocols.ByteReader):
+        """Create a new file."""
+        try:
+            descriptors = self._get_path_descriptors(to_ignore=-1)
```
[the remark this answers is not in the document]
Comment (capitancambio) [reply src/tentaclio/clients/google_drive_client.py:187]: yeah you're right, I'll think of another name

File: src/tentaclio/clients/google_drive_client.py [hunk 3]
```diff
@@ -180,7 +180,10 @@ def put(self, reader: protocols.ByteReader, **kwargs) -> None:
     @decorators.check_conn
     def scandir(self, **kwargs) -> Iterable[fs.DirEntry]:
         """List contents of a folder from google drive."""
-        leaf_descriptor = self._get_leaf_descriptor()
+        try:
+            leaf_descriptor = self._get_leaf_descriptor()
+        except DescriptorNotFound as e:
+            raise IOError(f"{self.url} not found.", e)
```
[the remark this answers is not in the document]
Comment (capitancambio) [reply src/tentaclio/clients/google_drive_client.py]: Yeah, you're right, I was feeling uneasy with how the exceptions were being thrown everywhere, I'll rethink this bit.

File: src/tentaclio/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (capitancambio) [reply src/tentaclio/__init__.py:59]: awesome catch.

**Comment by capitancambio** [comment]
I got rid of the `DescriptorException` all together @ali-tny, and just using `IOError` everywhere this simplifies the code quite a bit. ta!

**Pull request merged**